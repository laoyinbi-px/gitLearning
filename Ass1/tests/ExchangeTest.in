add trader1 100
add trader1 100
add trader2 -100
add trader2 100
add trader3 100
import trader1 ABC 100
buy trader2 ABC 20 7.5
SELL trader1 ABC 10 5
import trader1 book 100
export trader1 book 100
import trader1 book 100
SELL trader1 book 10 5
TRADES PRODUCT book
exit
