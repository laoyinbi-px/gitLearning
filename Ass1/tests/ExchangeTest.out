$ Success.
$ Trader with given ID already exists.
$ Initial balance cannot be negative.
$ Success.
$ Success.
$ Trader now has 100.00 units of ABC.
$ No trades could be made, order added to buy book.
$ Product sold in entirety, trades as follows:
trader1->trader2: 10.00xABC for $7.50.
$ Trader now has 100.00 units of book.
$ Trader now has no units of book.
$ Trader now has 100.00 units of book.
$ No trades could be made, order added to sell book.
$ No trades have been completed with given product.
$ Have a nice day.
